### PR TITLE
Fix 'parse_page' for French extractor

### PR DESCRIPTION
**Pull request by empiriker**
While exploring the progress to the French extractor you've made recently (well, done already!), I wanted to run the script from the default entry point via `wiktwords.py` but quickly discovered that there are still problems in `fr/page.py`.

One I fixed already: 
- https://github.com/tatuylonen/wiktextract/commit/fd53cdd1dfc5b3fc2a84595ab23cb113525ca1d5: Originally, empty dicts were created  instead of empty sets.

The main problem: 
- In several places, the code makes use of the inexistent instance method `find_content` on objects of the `WikiNode` class, for example in https://github.com/tatuylonen/wiktextract/blob/05321e53017a175dd2c7a293e1b4f784764a81f7/wiktextract/extractor/fr/page.py#L217 This seems intentional but I'am unsure what to do about it. @xxyzz Does this rely on some not yet merged feature?

For easier testing, I've added a simple test suite for the `parse_page` function of the French extractor.

ToDo:
- [x] replace or implement non existent `Wikinode.find_content()` method
- [x] run test suite successfully

### CODE REVIEW

**Comment by xxyzz** [comment]
`find_content()` is the method of `LevelNode` class: https://github.com/tatuylonen/wikitextprocessor/blob/main/wikitextprocessor/parser.py#L541

Those empty variables are just for compatible with old code in `page.py` and other files, they are not used for now.

Each subtitle sections are tested separately in their own test file, I think it's no harm to add a simple test for `parse_page()` but it shouldn't duplicate the existing tests.

File: tests/test_fr_page.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        self.wxr.wtp.close_db_conn()
+        close_thesaurus_db(
+            self.wxr.thesaurus_db_path, self.wxr.thesaurus_db_conn
+        )
+
+    def runpage(self, text):
+        assert isinstance(text, str)
+        return parse_page(self.wxr, self.wxr.wtp.title, text)
+
+    def test_page1(self):
```
Comment (xxyzz) [inline tests/test_fr_page.py:43]: I guess you copied the code from `tests/test_page.py` file, that file is not a good example... I have updated it: https://github.com/tatuylonen/wiktextract/commit/098461c5b1b60163fc19b603bbf67b2fc0de9eb6

And please use meaning name for test functions.

Comment (empiriker) [reply tests/test_fr_page.py:43]: Thanks for the review.

Yes, I didn't yet consider the final test form. I just wanted something quickly to run the parse_page function. Will improve!

**Comment by empiriker** [comment]
> `find_content()` is the method of `LevelNode` class: https://github.com/tatuylonen/wikitextprocessor/blob/main/wikitextprocessor/parser.py#L541
> 

@xxyzz Thanks for the pointer. That led me on the right path. My wikitextprocessor dependency was not up-to-date even though I ran the wiktextract installation again (`python -m pip install --use-pep517 -e ".[dev]"`). Since the version number of wikitextprocessor doesn't change it was satisfied with the old install. 

What's the best practice then to keep both libraries in sync (apart from deleting the python environment and reinstalling everything from scratch)?

**Comment by xxyzz** [comment]
I'd recommend install wiktextract first then force reinstall wikitextprocessor, both with the `-e` option and in the same venv.

**Comment by empiriker** [comment]
@xxyzz Just to clarify: You mean reinstall wikitextprocessor with the -e option from a local clone? Then I'd should just have to pull the most recent changes periodically.

**Comment by xxyzz** [comment]
Yes, clone both projects to your machine and install them with the `-e`(editable) option.

File: tests/test_fr_page.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        close_thesaurus_db(
+            self.wxr.thesaurus_db_path, self.wxr.thesaurus_db_conn
+        )
+
+    def mock_clean_node(self, *args, **kwargs):
+        try:
+            if args[2].template_name == "langue":
+                return "Français"
+        except:
+            return "Nom commun"
```
Comment (xxyzz) [inline tests/test_fr_page.py:42]: You could add these two templates using `Wtp.add_page()` and replace the page body with plain text(don't call lua code) before calling `parse_page()`.

File: tests/test_fr_page.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            self.wxr,
+            "exemple",
+            """
+== {{langue|fr}} ==
+=== {{S|nom|fr}} ===
+'''exemple'''
+""",
+        )
+        self.assertEqual(
+            [dict(d) for d in lst],  # Convert defaultdicts to dicts
```
Comment (xxyzz) [inline tests/test_fr_page.py:65]: No need to convert `defaultdicts`.

**Comment by empiriker** [comment]
I've updated the test file and now with some mocking and an up-to-date version of wikitextprocessor everything works.

The test doesn't cover much apart from that the parse_page function can be called correctly. 

That at least can catch error such as fixed by https://github.com/tatuylonen/wiktextract/pull/338/commits/d4540334051ba5395210d9aac57ac31f9a1624cd. Without properly declaring empty sets the assertion
` assert additional_expand is None or isinstance(additional_expand, set)`
[(Line 1852)](https://github.com/tatuylonen/wikitextprocessor/blob/9ac05aad22a5a828fd40f5ab5c29b172a328587b/wikitextprocessor/core.py#L1852)
would fail.

File: tests/test_fr_page.py (hunk 3 above)
Comment (empiriker) [reply tests/test_fr_page.py]: Thanks!

File: tests/test_fr_page.py (hunk 2 above)
Comment (empiriker) [reply tests/test_fr_page.py]: Yes, that's more elegant.

**Comment by empiriker** [comment]
By the way, do you prefer it that I use force-push (as is did so far) or that I create new commits while responding to your code reviews?

**Comment by xxyzz** [comment]
Force push is fine. And thanks for your contribution!
